filebeat/input/tcp/harvester_test.go

```diff
@@ -0,0 +1,266 @@
+package tcp
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/filebeat/harvester"
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type testingOutlet struct {
+	ch chan *util.Data
+}
+
+func (o *testingOutlet) OnEvent(data *util.Data) bool {
+	o.ch <- data
+	return true
+}
+
+func newTestingOutlet(ch chan *util.Data) *testingOutlet {
+	return &testingOutlet{ch: ch}
+}
+
+// This could be extracted into testing utils and we could add some unicode chars to the charsets.
+func randomString(l int) string {
```

Comment:
+1 for extracting this function

Reply:
I will wait when we add a bit more feature to it.